test/server/core/gameSteps/phases/SetupPhase.spec.ts

```diff
@@ -0,0 +1,100 @@
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                        ],
+                    },
+                    player2: {
+                        resources: [],
+                        deck: [
+                            'moisture-farmer',
+                            'atst',
+                            'atst',
+                            'atst',
+                            'atst',
+                            'atst',
+                            'atst',
+                            'atst',
+                            'wampa',
+                            'atst',
+                            'atst',
+                        ],
+                    }
+                });
+            });
+
+            it('should flow normally through each step', function () {
+                const { context } = contextRef;
+
+                // The setup phase is divided into 3 or 4 steps: choose Initiative player, Draw card step, mulligan step (optional), resource step
+                // Choose Initiative Step
+                context.selectInitiativePlayer(context.player1);
+
+                // Draw cards step
+                expect(context.player1.handSize).toBe(6);
+                expect(context.player2.handSize).toBe(6);
+                const beforeMulliganHand = context.player1.hand;
+
+                // Mulligan step
+                context.player1.clickPrompt('yes');
+                context.player2.clickPrompt('no');
+                const afterMulliganHand = context.player1.hand;
+                expect(beforeMulliganHand).not.toEqual(afterMulliganHand);
+
+                // Resource step
+                // check that no resource was automatically set
+                expect(context.player1.resources.length).toBe(0);
+                expect(context.player2.resources.length).toBe(0);
+
+                const player1FirstCard = context.player1.hand[0];
+                const player1SecondCard = context.player1.hand[1];
+
+                // select 2 cards to resource
+                context.player1.clickCard(context.player1.hand[0]);
+                context.player1.clickCard(context.player1.hand[1]);
+
+                // Check if selecting any unavailable cards triggers resourcing
+                context.player2.clickCardNonChecking(context.player2.deck[0]);
+                context.player2.clickCardNonChecking(context.player1.hand[0]);
+                context.player2.clickCardNonChecking(context.player1.deck[0]);
+                context.player2.clickCardNonChecking(context.p1Base);
+                context.player2.clickCardNonChecking(context.p2Base);
+
+                // Select 2 correct cards to resource
+                context.player2.clickCard(context.player2.hand[0]);
+                context.player2.clickCard(context.player2.hand[1]);
```

Comment:
So this is something I forgot - currently the setup phase resource prompt doesn't have a "Done" button, it just automatically resources once both players have clicked 2 cards. I think that's probably not what we want, it would be better to ask players to click "done" so they can lock in their choice.

Can you make that update to the ResourcePrompt? You can copy some of the logic from VariableResourcePrompt, or maybe even move it completely since ResourcePrompt is the parent class and VariableResourcePrompt will just inherit it.

Note that making this change will also require updating the test setup stuff since now the test setup will need to additionally click "Done" for each player

Reply:
sure i can add the Done to the resource phase.

Reply:
Agreed here. In general our approach is that multiple selection will require a confirmation while single selection automatically submits. Might revise that as we see more cases but certainly for this one we want a done button. Do we also have the ability to "un-resource" a card?

Reply:
The resourcing doesn't happen until the prompt resolves so shouldn't be needed, if I'm understanding the intent right

Reply:
if i understand @danbastin correctly its whether we support unselecting cards. If we don't, I can make an addition to the ResourcePrompts clickCard function to check whether the card is already in the selectedCards and remove it from there, if that makes sense.